src/main/java/com/gregtechceu/gtceu/common/data/GTBlocks.java

```diff
@@ -515,28 +536,36 @@ public static BlockEntry<Block> createCasingBlock(String name,
                                                       ResourceLocation texture,
                                                       NonNullSupplier<? extends Block> properties,
                                                       Supplier<Supplier<RenderType>> type) {
-        return REGISTRATE.block(name, blockSupplier)
+        var builder = REGISTRATE.block(name, blockSupplier)
                 .initialProperties(properties)
                 .properties(p -> p.isValidSpawn((state, level, pos, ent) -> false))
                 .addLayer(type)
                 .blockstate(GTModels.cubeAllModel(name, texture))
-                .tag(GTToolType.WRENCH.harvestTags.get(0), BlockTags.MINEABLE_WITH_PICKAXE)
+                .tag(GTToolType.WRENCH.harvestTags.get(0));
+        if (!ConfigHolder.INSTANCE.machines.requireGTToolsForBlocks) {
+            builder.tag(BlockTags.MINEABLE_WITH_PICKAXE);
+        }
+        return builder
                 .item(BlockItem::new)
                 .build()
                 .register();
     }
 
     private static BlockEntry<Block> createMachineCasingBlock(int tier) {
         String tierName = GTValues.VN[tier].toLowerCase(Locale.ROOT);
-        BlockEntry<Block> entry = REGISTRATE
+        var builder = REGISTRATE
                 .block("%s_machine_casing".formatted(tierName), Block::new)
                 .lang("%s Machine Casing".formatted(GTValues.VN[tier]))
                 .initialProperties(() -> Blocks.IRON_BLOCK)
                 .properties(p -> p.isValidSpawn((state, level, pos, ent) -> false))
                 .addLayer(() -> RenderType::cutoutMipped)
                 .blockstate(GTModels.createMachineCasingModel(tierName))
-                .tag(GTToolType.WRENCH.harvestTags.get(0), BlockTags.MINEABLE_WITH_PICKAXE)
-                .item(BlockItem::new)
+                .tag(GTToolType.WRENCH.harvestTags.get(0));
+        if (!ConfigHolder.INSTANCE.machines.requireGTToolsForBlocks) {
+            builder.tag(BlockTags.MINEABLE_WITH_PICKAXE);
+        }
```

Comment:
this won't work for removing the pickaxe tags as it's only applied when you run datagen.